### DISCUSSION TITLE
Run GitHub actions CI workflow on another users repo

### DISCUSSION THREAD

**Discussion by benstear** [community/community#27020] [Actions]
<p>Hi,</p>
<p>I am working on a project where I need to run a github actions CI workflow on other github users’ open repositories.  I am trying to gather python packages (with the authors’ permission) that are related to a specific scientific field. I need to run some basic tests (linting, build, unit tests) and if the workflow passes I will feature the python package on our website. </p>
<p>Is the only way to do this to clone the repo, add my workflow and run the tests? Or is there a more direct way to run a CI workflow on a repo that you do not own? </p>
<p>Thanks!</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/benstear">@benstear</a> , </p>
<p>In your workflow ,  you could use checkout action to checkout other repos beyond the current repo. </p>
<pre><code>- uses: actions/checkout@v2
      with:
        repository: orgname/reponame
</code></pre>

**Comment by benstear** [comment]
<p>That is much easier, thank you!</p>

**Comment by benstear** [comment]
<p>Is it possible to checkout a single file from a repo? I tried:</p>
<pre><code>- name: checkout repo
      uses: actions/checkout@v2
      with:
        repository: username/reponame

  - name: checkout single file
     uses: actions/checkout@v2
     with:
       repository: another_user/another_repo/directory/file.sh
</code></pre>
<p>But it looks like this is not supported in the checkout action. </p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/benstear">@benstear</a> </p>
<p>Checkout action doesn’t support checkout a single file, you could use wget to download a single file from github. </p>
<pre><code>wget https://raw.githubusercontent.com/username/reponame/branch/path/to/file
</code></pre>
